src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala

```diff
@@ -1,28 +1,49 @@
 package com.codacy.client.bitbucket.v2.service
 
+import com.codacy.client.bitbucket.client.{BitbucketClient, PageRequest, Request, RequestResponse}
 import com.codacy.client.bitbucket.v2.{Team, UserPermission}
-import com.codacy.client.bitbucket.client.{BitbucketClient, Request, RequestResponse}
 
 class TeamServices(client: BitbucketClient) {
 
   private val BaseUrl = "https://bitbucket.org/api/2.0/teams"
 
-  def list(role: String = "member"): RequestResponse[Seq[Team]] = {
-    client.executePaginated(Request(s"$BaseUrl?role=$role", classOf[Seq[Team]]))
+  def list(role: String = "member", pageRequest: Option[PageRequest] = None): RequestResponse[Seq[Team]] = {
+    pageRequest match {
+      case Some(request) =>
+        request.cursor match {
```

Comment:
We can do our executeWithCursor with this signature client.executeWithCursor(url, "class", PageRequest).

The advantage would be, that we would do the "pageRequest.cursor" logic of calling url or cursor only once